sdks/python/apache_beam/ml/inference/tensorrt_inference.py

```diff
@@ -0,0 +1,295 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# pytype: skip-file
+
+from __future__ import annotations
+
+import logging
+import sys
+from typing import Any
+from typing import Dict
+from typing import Iterable
+from typing import Optional
+from typing import Sequence
+from typing import Tuple
+
+import numpy as np
+
+from apache_beam.io.filesystems import FileSystems
+from apache_beam.ml.inference.base import ModelHandler
+from apache_beam.ml.inference.base import PredictionResult
+
+LOGGER = logging.getLogger("TensorRTEngineHandlerNumPy")
+# This try/catch block allows users to submit jobs from a machine without
+# GPU and other dependencies (tensorrt, cuda, etc.) at job submission time.
+try:
+  import tensorrt as trt
+  TRT_LOGGER = trt.Logger(trt.Logger.INFO)
+  trt.init_libnvinfer_plugins(TRT_LOGGER, namespace="")
+  LOGGER.info('tensorrt module successfully imported.')
+except ModuleNotFoundError:
+  TRT_LOGGER = None
+  msg = 'tensorrt module was not found. This is ok as long as the specified ' \
+    'runner has tensorrt dependencies installed.'
+  LOGGER.warning(msg)
+
+
+def _load_engine(engine_path):
+  import tensorrt as trt
+  file = FileSystems.open(engine_path, 'rb')
+  runtime = trt.Runtime(TRT_LOGGER)
+  engine = runtime.deserialize_cuda_engine(file.read())
+  assert engine
+  return engine
+
+
+def _load_onnx(onnx_path):
+  import tensorrt as trt
+  builder = trt.Builder(TRT_LOGGER)
+  network = builder.create_network(
+      flags=1 << int(trt.NetworkDefinitionCreationFlag.EXPLICIT_BATCH))
+  parser = trt.OnnxParser(network, TRT_LOGGER)
+  with FileSystems.open(onnx_path) as f:
+    if not parser.parse(f.read()):
+      LOGGER.error("Failed to load ONNX file: %s", onnx_path)
+      for error in range(parser.num_errors):
+        LOGGER.error(parser.get_error(error))
+      sys.exit(1)
+  return network, builder
+
+
+def _build_engine(network, builder):
+  import tensorrt as trt
+  config = builder.create_builder_config()
+  runtime = trt.Runtime(TRT_LOGGER)
+  plan = builder.build_serialized_network(network, config)
+  engine = runtime.deserialize_cuda_engine(plan)
+  builder.reset()
+  return engine
+
+
+def _validate_inference_args(inference_args):
```

Comment:
This method is no longer needed so you can clean this code up.

https://github.com/apache/beam/pull/22282

This is now validated in the base implementation.

Reply:
Ok, I'll take it out

Reply:
Addressed